server/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java

```diff
@@ -690,4 +696,56 @@ public XContentBuilder toXContent(final XContentBuilder builder, final Params pa
 
     }
 
+    static class IngestStats implements ToXContentFragment {
+
+        final int pipelineCount;
+        final Map<String, long[]> stats;
+
+        IngestStats(final List<NodeStats> nodeStats) {
+            Set<String> pipelineIds = new HashSet<>();
+            SortedMap<String, long[]> stats = new TreeMap<>();
+            for (NodeStats nodeStat : nodeStats) {
+                if (nodeStat.getIngestStats() != null) {
+                    for (Map.Entry<String,
+                            List<org.elasticsearch.ingest.IngestStats.ProcessorStat>> processorStats : nodeStat.getIngestStats()
+                            .getProcessorStats().entrySet()) {
+                        pipelineIds.add(processorStats.getKey());
+                        for (org.elasticsearch.ingest.IngestStats.ProcessorStat stat : processorStats.getValue()) {
+                            stats.compute(stat.getType(), (k, v) -> {
+                                if (v == null) {
```

Comment:
should we also add `ingestTimeInMillis` here?

Reply:
hmm. I suppose you are right. I was only interested in adding fields that I already know usages of, but in order to be more complete and have feature-parity with the node stats API, both `ingestTimeInMillis` and `ingestCurrent` should be added. is that what you mean?

Reply:
I think `ingestTimeInMillis` is an interesting metric to capture.
I'm not so sure about `ingestCurrent`. In any way I think for feature parity it is good to capture both.

Reply:
agreed. thanks for bringing it up. I'll update to include these

Reply:
updated. also found a mismatch in the naming of the `failed` field. it was `failed_count` in this PR for cluster stats, while `failed` in node stats.